sound/soc/sof/ipc4-topology.c

```diff
@@ -111,6 +111,42 @@ static const struct sof_topology_token gain_tokens[] = {
 		get_token_u32, offsetof(struct sof_ipc4_gain_data, init_val)},
 };
 
+struct sof_process_types {
+	const char *name;
+	enum sof_ipc_process_type type;
+	enum sof_comp_type comp_type;
+};
+
+static const struct sof_process_types sof_process[] = {
+	{ },
+};
+
+static enum sof_ipc_process_type find_process(const char *name)
+{
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(sof_process); i++) {
+		if (strcmp(name, sof_process[i].name) == 0)
+			return sof_process[i].type;
+	}
+
+	return SOF_PROCESS_NONE;
+}
+
+static int get_token_process_type(void *elem, void *object, u32 offset)
+{
+	u32 *val = (u32 *)((u8 *)object + offset);
+
+	*val = find_process((const char *)elem);
+	return 0;
+}
```

Comment:
@ujfalusi As @ranj063 recommended, I will use UUID instead of process_type. So these code will be removed in the next update. For sharing code, I tried to share these functions between ipc3 and ipc4. But find_process() needs a static variable `sof_process`, which is defined in ipc3-topology.c/ipc4-topology.c. This means the header file depends on `.c` file. I'm not feel very comfortable with this.  And another solution is we pass `sof_process` in get_token_process_type() as a parameter, but this need a big change in token parsing. So I copied the code directly.